ambry-store/src/test/java/com.github.ambry.store/IndexValueTest.java

```diff
@@ -190,8 +205,18 @@ private void verifyIndexValue(IndexValue value, String logSegmentName, long size
       long expiresAtMs, long originalMessageOffset, long operationTimeInMs, short serviceId, short containerId) {
     verifyGetters(value, logSegmentName, size, offset, isDeleted, expiresAtMs, originalMessageOffset, operationTimeInMs,
         serviceId, containerId);
+    // serialize and deserialize might change the value of expiry for version1
+    long expectedExpiryValue = -1;
+    switch (version) {
```

Comment:
nope. this is specifically for deserilization case. Before serialization and deser, negative expiration values are equivalent to what the user sets. But after ser and deser, any value < -1 is set to -1 and hence this code block.